### ISSUE TITLE
CamelCase warning of derive macro on client code

### ISSUE THREAD

**Issue by zeenix** [rust-lang/rust-analyzer#15541]
My crate uses `serde-repr` and the macros it provides doesn't conform to the CamelCase convention. However, since the macros are not mine, I don't think I should be given the warning about it. Here is my code:

```rust
use serde_repr::{Deserialize_repr, Serialize_repr};

#[repr(u8)]
#[derive(Debug, Deserialize_repr, Serialize_repr)]
pub enum EndianSig {
    /// The D-Bus message is in big-endian (network) byte order.
    Big = b'B',

    /// The D-Bus message is in little-endian byte order.
    Little = b'l',
}
```

and it gives me:

```rust
Structure `discriminant` should have CamelCase name, e.g. `Discriminant`rust-analyzer[non_camel_case_types](https://doc.rust-lang.org/rustc/?search%3Dnon_camel_case_types)
core::fmt::macros
macro Debug
Derive macro generating an impl of the trait Debug.
```

`cargo build` , `cargo check` or `cargo clippy` don't complain about this at all so I think RA shouldn't either.

**Comment by lnicola** [comment]
Duplicate of #15344.
